xbmc/addons/RepositoryUpdater.cpp

```diff
@@ -322,32 +326,29 @@ void CRepositoryUpdater::ScheduleUpdate(UpdateScheduleType scheduleType)
   if (!m_addonMgr.HasAddons(AddonType::REPOSITORY))
     return;
 
-  int delta{1};
+  milliseconds delta{1};
   const auto nextCheck = ClosestNextCheck();
   if (nextCheck.IsValid())
   {
-    // Repos were already checked once and we know when to check next
-    delta = std::max(1, (nextCheck - CDateTime::GetCurrentDateTime()).GetSecondsTotal() * 1000);
-    CLog::Log(LOGDEBUG, "CRepositoryUpdater: closest next update check at {} (in {} s)",
-              nextCheck.GetAsLocalizedDateTime(), delta / 1000);
+    // Repos were already checked once and we know when to check next.
+    // delta must be positive and not zero (m_timer.Start() ignores 0 wait time)
+    delta = std::max<milliseconds>(
+        delta, seconds((nextCheck - CDateTime::GetCurrentDateTime()).GetSecondsTotal()));
+    CLog::Log(LOGDEBUG, "CRepositoryUpdater: closest next update check at {} (in {})",
+              nextCheck.GetAsLocalizedDateTime(), duration_cast<seconds>(delta));
   }
 
   if (scheduleType == UpdateScheduleType::Regular)
   {
     // Enforce minimum hold-off time of 1 hour between regular updates - this is especially
     // important to handle all sorts of failure cases (e.g., failure to update the add-on database)
     // that would otherwise lead to an immediate new update attempt and continuous hammering of the servers.
-    delta = std::max(1 * 60 * 60 * 1'000, delta);
-  }
-  else
-  {
-    // delta must be positive and not zero (m_timer.Start() ignores 0 wait time)
-    delta = std::max(1, delta);
+    delta = std::max<milliseconds>(hours(1), delta);
   }
 
-  CLog::Log(LOGDEBUG, "CRepositoryUpdater: checking in {} ms", delta);
+  CLog::Log(LOGDEBUG, "CRepositoryUpdater: checking in {}", delta);
```

Comment:
Just nitpicking, but I would keep the "ms" so the log reader knows the unit of measurement of the value.

Reply:
The unit is automatically added by the formatter for `std::chrono` types :slightly_smiling_face: